flow/layers/shader_mask_layer.cc

```diff
@@ -60,8 +60,10 @@ void ShaderMaskLayer::Paint(PaintContext& context) const {
   PaintChildren(context);
 
   SkPaint paint;
-  paint.setBlendMode(blend_mode_);
-  paint.setShader(shader_);
+  paint.setBlendMode(ToSk(blend_mode_));
+  if (shader_) {
+    paint.setShader(shader_->skia_object());
+  }
```

Comment:
This is a case where we might want to have "if (context.leaf_nodes_builder) {" and then do this natively in the builder language to avoid loss of information. I think CFLayer is "builder aware" as an example.

Reply:
Filed https://github.com/flutter/flutter/issues/109812